File: framework/languages/dotnet/e2e/RegistrationCodec/Server/JsonOnlyPeer/ServerOptions.cs
namespace RegistrationCodec.Server.JsonOnlyPeer;

using Zlink.Framework.E2E.Configuration;

public sealed record ServerOptions(
    string Rid,
    string HttpUrl,
    string LogDir,
    string? ChannelEndpoint,
    string? EvidenceFile,
    string? InvalidMode,
    string CodecMode,
    string? JsonOnlyPeerProject)
{
    public static ServerOptions Parse(string[] args)
        => E2eConfiguration.Load<ServerOptions>(args);
}

File: framework/languages/java/e2e/RegistrationCodec/Server/InvalidDuplicate/src/main/java/systems/zlink/e2e/registrationcodec/invalidduplicate/Configuration/ServerOptions.java
package systems.zlink.e2e.registrationcodec.invalidduplicate.Configuration;

public record ServerOptions(String serverEndpoint) {
    public static ServerOptions fromEnv() {
        return new ServerOptions(Env.get("ZLINK_JAVA_E2E_SERVER_ENDPOINT"));
    }
}
